src/release/breaking-changes/insert-content-text-input-client.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Insert Content Text Input Client
+description: Add a new method to the TextInputClient interface to allow Android virtual keyboards to
+insert rich content into Flutter TextFields. 
+---
+
+## Summary
+
+Adds `insertContent` method to the `TextInputClient` interface to allow Android's image keyboard 
+feature to insert content into a Flutter `TextField`.
+
+
+## Context
+
+As of Android 7.1, IMEs (virtual keyboards) are able to directly send images and rich content
+into a text editor. This allows users to directly insert gifs, stickers, or context-aware rich
+content into a text field.
+
+
+## Description of change
+
+When the user inserts rich content via the IME, the platform sends a `commitContent` channel message
+called `TextInputClient.commitContent`.
+This notifies the Dart code that rich content was inserted. 
+The channel message contains the mime type, URI, and bytedata for the inserted content in JSON form.
+
+
+## Migration guide
+
+If you previously implemented `TextInputClient`, you must override
+`insertContent` to either support rich content insertion or provide an empty implementation.
+
+To migrate, implement `insertContent`.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+class MyCustomTextInputClient implements TextInputClient {
+  ...
+}
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+class MyCustomTextInputClient implements TextInputClient {
+  ...
+  @override
+  void insertContent() {
+    ...
+  }
+  ...
+}
+```
```

Comment:
Some ideas of other things we might want to add:

  1. Maybe this is obvious, but it might help to state that if their implementation of TextInputClient doesn't need the ability to receive image data from Android virtual keyboards, then they can feel free to leave the implementation empty.

```dart
  @override
  void insertContent() {}
```
  2. State that instead of adding an empty implementation, if they don't want to be broken by changes like this in the future, they can use TextInputClient with the `with` keyword instead of `implements`.
  3. Maybe we should direct them to the implementation in TextInputClient as an example.  Like "TextInputClient calls simply calls a callback when `insertContent` is called.  See <link to code on GitHub when it's merged> for more details."